src/tools/rbd_mirror/image_sync/SnapshotCopyRequest.cc

```diff
@@ -27,11 +27,11 @@ const std::string &get_snapshot_name(I *image_ctx, librados::snap_t snap_id) {
   auto snap_it = std::find_if(image_ctx->snap_ids.begin(),
                               image_ctx->snap_ids.end(),
                               [snap_id](
-      const std::pair<std::string, librados::snap_t> &pair) {
+      const std::pair<std::pair<cls::rbd::SnapshotNamespace, std::string>, librados::snap_t> &pair) {
```

Comment:
It looks like auto in lambdas produces a warning unless I specify parameter -std=c++1y. Do you prefer the warning or just wrap the lines?